src/Microsoft.Azure.SignalR/EndpointRouters/DefaultEndpointRouter.cs

```diff
@@ -45,5 +46,49 @@ private ServiceEndpoint[] GetNegotiateEndpoints(IEnumerable<ServiceEndpoint> end
 
             return secondary;
         }
+
+        /// <summary>
+        ///  Choose endpoint randomly by weight. 
+        ///  The weight is defined as the remaining connection quota.
+        ///  The least weight is set to 1. So instance with no connection quota still has chance.
+        /// </summary>
+        private ServiceEndpoint GetEndpointAccordingToWeight(ServiceEndpoint[] availableEndpoints)
+        {
+            //first check if weight is available or necessary
+            if (availableEndpoints.Any(endpoint => endpoint.EndpointMetrics.ConnectionCapacity == 0) ||
```

Comment:
Does `Capacity==0` mean status is not initiated? Should we random pick only when `All` instead of `Any` are not initiated.

Reply:
Once runtime is up, it's inited.  Capacity==0 happens when it's free or it's old version runtime.

Reply:
Why free doesn't have capacity? Even 20 is a valid number.
And should a free one break the weight algorithm always?

Reply:
The Capacity might be 0 when no ping has been received. That happens when server starts up. Since no ping for some time will lead to a connection drop. That will be a short time. Using `any` just to explicitly show some backward compatity though removing that won't have difference.